src/lib/core/ripple/ripple-renderer.ts

```diff
@@ -175,21 +187,37 @@ export class RippleRenderer {
     this._triggerElement = element;
   }
 
-  /** Function being called whenever the trigger is being pressed. */
+  /** Function being called whenever the trigger is being pressed using mouse. */
   private onMousedown(event: MouseEvent) {
-    if (!this.rippleDisabled) {
-      this._isMousedown = true;
+    const isSyntheticEvent = this._lastTouchStartEvent &&
+        Date.now() < this._lastTouchStartEvent + ignoreMouseEventsTimeout;
```

Comment:
Rather than using the `Date.now()`, you should be able to take the `Event.timeStamp`. I'm not completely sure about the browser support though, you may have to double check.

Reply:
As discussed on Slack, for browser compatibility we'll be using `Date.now()`